### PR TITLE
feat: validate sync records

### PR DESCRIPTION
**Pull request by bodinsamuel**
## Describe your changes

Fixes https://linear.app/nango/issue/NAN-1156/use-json-schema-to-validate-records-at-run-time
Fixes https://linear.app/nango/issue/NAN-1347/flag-in-cli-to-bypass-validation
Fixes https://linear.app/nango/issue/NAN-1348/improve-cli-display

- Validate records 
- Add flag to bypass validation in the CLI
- Better CLI output for validation

## Test

```sh
node packages/cli/dist/index.js dryrun __SYNC__ __CONNECTION__ --no-validation
```
Screenshot 2024-07-04 at 15 29 46

### CODE REVIEW

**Comment by khaliqgant** [comment]
@bodinsamuel let me know when this is ready for review (with build issues resolved)

**Comment by bodinsamuel** [comment]
arg yes, fixing that thanks

**Comment by bodinsamuel** [comment]
@khaliqgant all good 🙏🏻

**Comment by bodinsamuel** [comment]
After our discussion, I made the flag optin right now. It's will log an error but only halt execution if `--validation` is passed

**Review by khaliqgant — APPROVED** [review]

File: packages/cli/lib/services/dryrun.service.ts [hunk 1]
```diff
@@ -293,7 +306,13 @@ export class DryRunService {
             loadLocation: './',
             debug,
             logMessages,
-            stubbedMetadata
+            stubbedMetadata,
+            runnerFlags: {
+                validateActionInput: this.validation, // irrelevant for cli
+                validateActionOutput: this.validation, // irrelevant for cli
```
Comment (TBonnin) [inline packages/cli/lib/services/dryrun.service.ts:312]: why are we passing them if there are irrelevant?

**Pull request merged**